app/assets/javascripts/geoblacklight/modules/relations.js.erb

```diff
@@ -0,0 +1,80 @@
+var setupDataRelations = function () {
+    $('#sidebar').append($('<div class="panel panel-default show-tools" id="relation_tools"><div class="panel-heading">Data Relations</div><div class="panel-body"><ul class="nav"><div class="ancestor_links"></div><div class="descendant_links"></div></ul></div></div>').hide().fadeIn(200));
+}
+
+var setupAncestorLinks = function (count) {
+    $('#relation_tools .nav').append('<b>Source Datasets (' + count + ')</b>');
+}
+
+var setupDescendantLinks = function (count) {
+    $('#relation_tools .nav').append('<b>Derived Datasets (' + count + ')</b>');
+}
+
+var populateAncestorLinks = function (title, slug, baseurl) {
+    $('#relation_tools .nav').append('<li><a href="' + urlify_record(baseurl, slug) + '"><span class="geoblacklight geoblacklight-relations-ancestor"> </span> ' + title + '</a></li>');
+}
+
+var populateDescendantLinks = function (title, slug, baseurl) {
+    $('#relation_tools .nav').append('<li><a href="' + urlify_record(baseurl, slug) + '"><span class="geoblacklight geoblacklight-relations-descendant"> </span> ' + title + '</a></li>');
+}
+
+var populateDescendantSearchLink = function (ancestor_slug, count, baseurl) {
+    $('#relation_tools .nav').append('<li><a href="' + urlify_search(baseurl, ancestor_slug) + '">Browse all ' + count + ' items...</a></li>');
+}
+
+var urlify_search = function (site_url, slug) {
+    return site_url + '?f[<%= Settings.FIELDS.SOURCE %>][]=' + slug
+}
+var urlify_record = function (site_url, slug) {
+    return site_url + 'catalog/' + slug;
+}
+
+var grab_document_url = function () {
```

Comment:
rather than having this get from the document, can we provide in a data-attribute?